### PR TITLE
Newclasses

### PR DESCRIPTION
**Pull request by bayne-grubb**
Added a bunch o new classes and modified the universe generation code to make unique names as well as multiple planets in a solar system.

### CODE REVIEW

File: app/src/main/java/com/communistutopia/spacetrader/model/Universe.kt [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+    fun generateUniverse() {
+        var i: Int = NUM_SYSTEMS
+        while (i > 0) {
+            val x = random.nextInt(X_SIZE)
+            val y = random.nextInt(Y_SIZE)
+            val num_planets = random.nextInt(MAX_PLANETS) + 1
+            val pair = Pair(x, y)
+            if (!universe.containsKey(pair)) {
+                val planetSet = generatePlanets(num_planets)
+                val systemName: String = namesList[random.nextInt(namesList.size)]
```
Comment (haleylittmann) [inline app/src/main/java/com/communistutopia/spacetrader/model/Universe.kt:183]: could you still get a solar system with the same names or a planet and solar system with the same name? I don't see where the solar system's name is also removed afterwards

Comment (bayne-grubb) [reply app/src/main/java/com/communistutopia/spacetrader/model/Universe.kt:183]: yes, fixing it now

**Review by rohanrk — CHANGES_REQUESTED** [review]
Could you add logic to print the Universe to the logcat?

**Comment by bayne-grubb** [comment]
> Could you add logic to print the Universe to the logcat?

Drake is going to do this in his UI implementation, I'll close this for now

**Pull request merged**